### PR TITLE
Added build with prebuilt libs

### PR DESCRIPTION
**Pull request by MarnixKuijs**
Added the capability for building ash-molten with prebuilt libs of MoltenVK.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -118,22 +133,46 @@ fn main() {
...
+                std::env::var("CARGO_CFG_TARGET_OS").unwrap(),
+                std::env::var("CARGO_CFG_TARGET_ARCH").unwrap()
+            ));
+
             pb
         };
+
+        println!("cargo:rustc-link-search=native={}", project_dir.display());
+    } else {
+        let target_dir = Path::new(&std::env::var("OUT_DIR").unwrap())
```
Comment (repi) [inline build.rs:157]: so this repurposes to existing `external` feature and re-uses it to download a pre-built library right? but that also means you can't provide your own if you have it (which is what the previous `external` feature meant).

May be cleanest to have 2 different features for this, like maybe `external-provided` and `external-downloaded` or something like that?

The `external-downloaded` and this case we should also add a separate CI build step that builds with that feature so we make sure it continues to work

**Review by repi — CHANGES_REQUESTED** [review]
Nice!

Some notes

File: build.rs [hunk 2]
```diff
@@ -110,6 +110,21 @@ mod mac {
         handle.join().unwrap();
         target_name
     }
+
+    pub(crate) fn download_prebuilt_molten<P: AsRef<Path>>(target_dir: &P) {
+        use std::process::Command;
+
+        let status = Command::new("sh")
+            .arg("download-prebuilt.sh")
```
[the remark this answers is not in the document]
Comment (MarnixKuijs) [reply build.rs:118]: Fair enough. I will just get rid of the .sh and just add commands to the build.rs

File: build.rs (hunk 1 above)
Comment (MarnixKuijs) [reply build.rs:157]: Good point. I will add a additional feature.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -25,7 +25,11 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all-features -- -D warnings
+          args: --features external -- -D warnings
```
Comment (repi) [inline .github/workflows/ci.yaml:28]: This should just build the default feature also right? As CI can't build with the `external` feature as it doesn't have an externally provided MoltenVK?

```suggestion
          args: -- -D warnings
```

**Review by repi — CHANGES_REQUESTED** [review]
Tested locally and worked fine.

This looks good also, one small catch.

Also do think should add some brief comments about what the features do (including the default one) in `Cargo.toml` and also to README of the different modes, and some instructions on how to upload a new prebuilt MoltenVK library

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (MarnixKuijs) [reply .github/workflows/ci.yaml:28]: Ooh oops your are right

**Comment by MarnixKuijs** [comment]
> Also do think should add some brief comments about what the features do (including the default one) in `Cargo.toml` and also to README of the different modes, and some instructions on how to upload a new prebuilt MoltenVK library

Good point I will add some

**Pull request merged**